addons/rating/models/rating.py

```diff
@@ -32,16 +32,27 @@ def new_access_token(self):
     res_model_id = fields.Many2one('ir.model', 'Related Document Model', index=True, ondelete='cascade', help='Model of the followed resource')
     res_model = fields.Char(string='Document Model', related='res_model_id.model', store=True, index=True, readonly=True)
     res_id = fields.Integer(string='Document ID', required=True, help="Identifier of the rated object", index=True)
+    parent_res_name = fields.Char(compute='_compute_parent_res_name', String='Parent Document Name', store=True)
+    parent_res_model_id = fields.Many2one('ir.model', 'Parent Related Document Model', index=True)
+    parent_res_model = fields.Char(string='Parent Document Model', store=True, related='parent_res_model_id.model', index=True)
+    parent_res_id = fields.Integer(string='Parent Document ID', index=True)
     rated_partner_id = fields.Many2one('res.partner', string="Rated person", help="Owner of the rated resource")
     partner_id = fields.Many2one('res.partner', string='Customer', help="Author of the rating")
     rating = fields.Float(string="Rating", group_operator="avg", default=0, help="Rating value: 0=Unhappy, 10=Happy")
     rating_image = fields.Binary('Image', compute='_compute_rating_image')
-    rating_text = fields.Char(string='Rating', compute='_compute_rating_text')
+    rating_text = fields.Char(string='Rating', store=True, compute='_compute_rating_text')
```

Comment:
You cannot store translated terms.
In a separated commit, try to group on rating_text, but change this field into a selection computed field (readonly)